aten/src/ATen/native/vulkan/api/Context.cpp

```diff
@@ -42,8 +42,12 @@ VkDevice create_device(
       nullptr,
       &device_extension_properties_count,
       device_extension_properties.data()));
-
+  #ifdef _MSC_VER
+  // MSVC does not support VLAs (variable length arrays).
+  std::vector<const char*> requested_device_extensions{
+  #else
   constexpr const char* const requested_device_extensions[]{
+  #endif
```

Comment:
Let compiler figure out the best type (I suspect it would be [std::initializer_list](https://en.cppreference.com/w/cpp/utility/initializer_list))
```suggestion
  constexpr auto requested_device_extensions = {
  #endif
```